framework/include/cppmicroservices/ListenerFunctors.h

```diff
@@ -24,15 +24,69 @@
 #define CPPMICROSERVICES_LISTENERFUNCTORS_H
 
 #include "cppmicroservices/GlobalConfig.h"
+#include "cppmicroservices/FrameworkExport.h"
 
 #include <cstring>
 #include <functional>
+#include <cstdint>
 
 namespace cppmicroservices {
 
   class ServiceEvent;
   class BundleEvent;
   class FrameworkEvent;
+  class ServiceListeners;
+
+  typedef std::uint64_t ListenerTokenId;
+
+  /**
+   * \brief The token returned when a listener is registered with BundleContext.
+   *
+   * The token object enables the clients to remove the listeners from the BundleContext.
+   * This is a move-only type, with the intention that the transfer of ownership will
+   * be handled explicitly by the clients.
+   *
+   */
+  class US_Framework_EXPORT ListenerToken
```

Comment:
The way I tackled both this and the suggestion below is to 
- separate out the ListenerToken declaration into ListenerToken.h and export it
- rename ListenerFunctors.cpp to ListenerToken.cpp (where the definition of ListenerToken resides)